src/AbstractSchemaReader.php

```diff
@@ -0,0 +1,760 @@
+    *
+    * @return SchemaItem
+    */
+    abstract protected function findSomeType(
+        SchemaItem $fromThis,
+        DOMElement $node,
+        $attributeName
+    );
+
+    /**
+    * @param string $attributeName
+    *
+    * @return SchemaItem
+    */
+    abstract protected function findSomeTypeFromAttribute(
+        SchemaItem $fromThis,
+        DOMElement $node,
+        $attributeName
+    );
+
+    abstract protected function loadUnion(SimpleType $type, DOMElement $node);
+
+    /**
+    * @param bool $checkAbstract
+    */
+    abstract protected function fillTypeNode(Type $type, DOMElement $node, $checkAbstract = false);
+
+    abstract protected function loadExtension(BaseComplexType $type, DOMElement $node);
+
+    public function findAndSetSomeBase(
+        Type $type,
+        Base $setBaseOnThis,
+        DOMElement $node
+    ) {
+        /**
+        * @var Type $parent
+        */
+        $parent = $this->findSomeType($type, $node, 'base');
+        $setBaseOnThis->setBase($parent);
+    }
+
+    abstract protected function maybeLoadExtensionFromBaseComplexType(
+        Type $type,
+        DOMElement $childNode
+    );
+
+    abstract protected function loadRestriction(Type $type, DOMElement $node);
+
+    /**
+    * @param string $typeName
+    *
+    * @return mixed[]
+    */
+    abstract protected static function splitParts(DOMElement $node, $typeName);
+
+    /**
+     *
+     * @param string $finder
+     * @param Schema $schema
+     * @param DOMElement $node
+     * @param string $typeName
+     * @throws TypeException
+     * @return ElementItem|Group|AttributeItem|AttributeGroup|Type
+     */
+    public function findSomething($finder, Schema $schema, DOMElement $node, $typeName)
+    {
+        list ($name, $namespace) = static::splitParts($node, $typeName);
+
+        /**
+        * @var string|null $namespace
+        */
+        $namespace = $namespace ?: $schema->getTargetNamespace();
+
+        try {
+            /**
+            * @var ElementItem|Group|AttributeItem|AttributeGroup|Type $out
+            */
+            $out = $schema->$finder($name, $namespace);
+
+            return $out;
+        } catch (TypeNotFoundException $e) {
+            throw new TypeException(sprintf("Can't find %s named {%s}#%s, at line %d in %s ", strtolower(substr($finder, 4)), $namespace, $name, $node->getLineNo(), $node->ownerDocument->documentURI), 0, $e);
+        }
+    }
+
+    /**
+    * @return Closure
+    */
+    abstract protected function loadElementDef(Schema $schema, DOMElement $node);
+
+    public function fillItem(Item $element, DOMElement $node)
+    {
+        /**
+        * @var bool $skip
+        */
+        $skip = false;
+        static::againstDOMNodeList(
+            $node,
+            function (
+                DOMElement $node,
+                DOMElement $childNode
+            ) use (
+                $element,
+                & $skip
+            ) {
+                if (
```

Comment:
In my opinion the code style used here is less readable than having it on the single line... is a single if statement split on almost 10 lines of code...

Reply:
force of habit; I prefer to have code readable on a 80-character terminal.

can we make a separate issue for editorconfig & automated php-cs-fixer stuff for code style?

Reply:
ok!